app/authorities/scholars_archive/cache_based_authority.rb

```diff
@@ -0,0 +1,20 @@
+module ScholarsArchive
+  class CacheBasedAuthority < Qa::Authorities::Local::FileBasedAuthority
+    private
+
+      def terms
+        authority_yaml = YAML.load(File.read(subauthority_filename))
+        uri = authority_yaml.with_indifferent_access.dig(@subauthority, :uri)
+        raise "#{@subauthority} uri configuration not found. " unless uri
+        result = ScholarsArchive::CachingService.fetch_or_store_in_cache(uri)
```

Comment:
would be ideal to grab the cache timeout from the config here and pass into the CachingService